orderer/common/multichannel/registrar.go

```diff
@@ -155,8 +160,57 @@ func NewRegistrar(
 func (r *Registrar) Initialize(consenters map[string]consensus.Consenter) {
 	r.consenters = consenters
 	existingChannels := r.ledgerFactory.ChannelIDs()
+	existingChannelsMap := map[string]struct{}{}
+	for _, channel := range existingChannels {
+		existingChannelsMap[channel] = struct{}{}
+	}
+
+	if r.config.ChannelParticipation.Enabled {
+		r.initializeFileRepos()
+
+		// check for existing joinblocks
+		joinBlockFiles, err := r.joinBlockFileRepo.List()
+		if err != nil {
+			logger.Panicf("Error listing files in joinblock file repo: %s", err)
+		}
+
+		for _, file := range joinBlockFiles {
+			channel := r.joinBlockFileRepo.FileToBaseName(file)
+
+			blockBytes, err := r.joinBlockFileRepo.Read(channel)
+			if err != nil {
+				logger.Panicf("Error reading joinblock %s: %s", file, err)
+			}
+			block := protoutil.UnmarshalBlockOrPanic(blockBytes)
+			configTx := protoutil.ExtractEnvelopeOrPanic(block, 0)
+
+			ledgerResources, err := r.newLedgerResources(configTx)
+			if err != nil {
+				logger.Panicf("Error creating ledger resources: %s", err)
+			}
+
+			chain, err := newChainSupportForJoin(
+				block,
+				r,
+				ledgerResources,
+				r.consenters,
+				r.signer,
+				r.blockcutterMetrics,
+				r.bccsp,
+			)
+			if err != nil {
+				logger.Panicf("Error creating chain support: %s", err)
+			}
+
+			r.chains[channel] = chain
+			chain.start()
 
```

Comment:
we should not start until we have all the chains initialized,  or we run the risk of a follower thread accessing the chains map before we finish populating it.

Reply:
This was handled via @tock-ibm 's refactor of the registrar initialization.